plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/remove/RemoveFromIndexPresenter.java

```diff
@@ -185,22 +185,20 @@ protected void onFailure(Throwable exception) {
     protected ResourceBasedNode<?> getResourceBasedNode() {
         List<?> selection = projectExplorer.getSelection().getAllElements();
         //we should be sure that user selected single element to work with it
-        if (selection != null && selection.isEmpty() || selection.size() > 1) {
-            return null;
-        }
+        if (selection != null && selection.size() > 1) {
 
-        Object o = selection.get(0);
+            Object o = selection.get(0);
 
-        if (o instanceof ResourceBasedNode<?>) {
-            ResourceBasedNode<?> node = (ResourceBasedNode<?>)o;
-            //it may be file node, so we should take parent node
-            if (node.isLeaf() && isResourceAndStorableNode(node.getParent())) {
-                return (ResourceBasedNode<?>)node.getParent();
-            }
+            if (o instanceof ResourceBasedNode<?>) {
+                ResourceBasedNode<?> node = (ResourceBasedNode<?>)o;
```

Comment:
i don't really know, because i changed condition of this stuff and only format it